File: LeetCode_Practice/LinkedLists/_141_LinkedListCycle.java
package LeetCode_Practice.LinkedLists;

class ListNode {
    int val;
    ListNode next;
    ListNode(int x) {
        val = x;
        next = null;
    }
}

public class _141_LinkedListCycle {
    public boolean hasCycle (ListNode head) {
        ListNode cur = head;
        if (head == null || head.next == null)
            return false;
        while (head != null && head.next != null) {
            cur = cur.next;
            head = head.next.next;
            if (cur == head)
                return true;
        }
        return false;
    }
}

File: LeetCode_Practice/C++_Solutions/LinkedLists/_141_LinkedListCycle.cpp
#include <iostream>

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution
{
    public:
        bool hasCycle (ListNode *head)
        {
            ListNode *cur = head;
            
            while (head != NULL && head->next != NULL)
            {
                cur = cur->next;
                head = head->next->next;
                if (cur == head)
                    return true;
            }
            return false;
        }
};

